### DISCUSSION TITLE
Issues with getting a Jetson Nano connected to GDS

### DISCUSSION THREAD

**Discussion by Janelletran22** [nasa/fprime#2603] [Q&A] [answered]
# Issue
I was running through the led-blinker tutorial (https://fprime-community.github.io/fprime-workshop-led-blinker/) and was able to get a deployment and GDS successfully running on an Odroid-M1 (got the green light on the top right and was able to send commands successfully).

However when I tried running the same deployment on a Jetson Nano, I was unnable to get the gds to run (red X on the top right).

After running the commands in the link below, I get these results:
(https://github.com/nasa/fprime/blob/devel/docs/Tutorials/CrossCompilationSetup/ArmLinuxTutorial.md)

- Deployment runs successfully on the Jetson Nano (doesn't crash)
- GDS command will run on my laptop, but gets stuck at "Listening for  ___" in the terminal (doesn't crash)
- actual GDS website shows red X at top right

## Configuration
- Windows laptop running with WSL
- Laptop connected to Jetson Nano with ethernet cable
- LED is not connected to the Jetson Nano (but I don't believe this would cause an issue)
- SSH'ed into the Jetson Nano through my laptop

## Things I tried/troubleshooted
- disabling firewall on the Jetson
- opening the Jetson port
- the same deployment will run on a beaglebone black
- tried making a new deployment without the gpio parts (to try to simplify the FSW) and didn't work either

# Thoughts
- FSW might be crashing?
- Something to do with the Jetson Nano's configuration might not be set up properly for FPrime?

Any tips and tricks with the Jetson Nano would be greatly appreciated!! Thank you!

**Comment by LeStarch** [comment]
Which TCP driver were you using in F´?  TcpClient or TcpServer?  My guess is it is a networking issue between the Jetson and the laptop.

For easiest results, you should be running with "TcpServer" in the flight software, and GDS in client mode `fprime-gds --ip-client`.  I suspect that since the GDS is "listening" this was not done.

**Reply by Janelletran22** [reply]
I'm currently running TcpServer. Weird thing is that I am able to run and connect FSW to the GDS using "fprime-gds --ip-client" successfully. But when trying to run GDS without running the native compilation using "fprime-gds -n --dictionary build-artifacts/aarch64-linux/<name-of-deployment>/dict/<App Dictionary>.xml --ip-client --ip-address <device-address>", I get stuck in the "Listening" loop on the GDS side, even with the deployment successfully running on the Jetson. Even with running with TcpServer, it still looks like a networking error?

**Reply by LeStarch** [reply]
When you run the deployment on the Nano, what address/port are you running?  For TCP server, you should use: 0.0.0.0 and any port (default 50000).  

```
./NanoDeployment -a 0.0.0.0 -p 50000
```

Then for the GDS you must supply the IP address of the Nano:

```
fprime-gds -n --dictionary build-artifacts/aarch64-linux//dict/.xml --ip-client --ip-address <ip address of nano> --ip-port 50000
```

If these are as above, and it does not work then there is an issue networking between your host and your board.

**Reply by Janelletran22** [reply]
Yep! Running the address 0.0.0.0 with the default port 50000. And also ran the GDS command with the Nano's ip address. I will try the same deployment on another Jetson to see if it's specifically the board's issue. Thank you for your help!

**Reply by LeStarch** [reply]
It could be a firewall issue too.  Make sure the Person's firewall is open on port 50000.

**Comment by Lex-ari** [comment, marked as answer]
A setting in the flight software and a possibly changed implementation in the kernel for Jetpack Linux may have been denying all TCP SYN requests. I made a PR here that should fix this issue #2785.
